receiver/memcachedreceiver/scraper.go

```diff
@@ -69,38 +65,133 @@ func (r *memcachedScraper) scrape(_ context.Context) (pdata.Metrics, error) {
+	totalConn := initMetric(ilm.Metrics(), metadata.M.MemcachedTotalConnections).Sum().DataPoints()
+	currItems := initMetric(ilm.Metrics(), metadata.M.MemcachedCurrentItems).Sum().DataPoints()
+	threads := initMetric(ilm.Metrics(), metadata.M.MemcachedThreads).Gauge().DataPoints()
+	evictions := initMetric(ilm.Metrics(), metadata.M.MemcachedEvictions).Sum().DataPoints()
+
 	for _, stats := range stats {
 		for k, v := range stats.Stats {
+			attributes := pdata.NewAttributeMap()
 			switch k {
 			case "bytes":
-				addIntGauge(ilm.Metrics(), metadata.M.MemcachedBytes.Init, now, parseInt(v))
+				addToIntMetric(bytes, attributes, parseInt(v), now)
 			case "curr_connections":
-				addIntGauge(ilm.Metrics(), metadata.M.MemcachedCurrentConnections.Init, now, parseInt(v))
+				addToIntMetric(currConn, attributes, parseInt(v), now)
 			case "total_connections":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedTotalConnections.Init, now, parseInt(v))
+				addToIntMetric(totalConn, attributes, parseInt(v), now)
+			case "cmd_get":
+				attributes.Insert(metadata.A.Command, pdata.NewAttributeValueString("get"))
+				addToIntMetric(commandCount, attributes, parseInt(v), now)
+			case "cmd_set":
+				attributes.Insert(metadata.A.Command, pdata.NewAttributeValueString("set"))
+				addToIntMetric(commandCount, attributes, parseInt(v), now)
+			case "cmd_flush":
+				attributes.Insert(metadata.A.Command, pdata.NewAttributeValueString("flush"))
+				addToIntMetric(commandCount, attributes, parseInt(v), now)
+			case "cmd_touch":
+				attributes.Insert(metadata.A.Command, pdata.NewAttributeValueString("touch"))
+				addToIntMetric(commandCount, attributes, parseInt(v), now)
+			case "curr_items":
+				addToDoubleMetric(currItems, attributes, parseFloat(v), now)
+			case "threads":
+				addToDoubleMetric(threads, attributes, parseFloat(v), now)
+			case "evictions":
+				addToIntMetric(evictions, attributes, parseInt(v), now)
+			case "bytes_read":
+				attributes.Insert(metadata.A.Direction, pdata.NewAttributeValueString("received"))
+				addToIntMetric(network, attributes, parseInt(v), now)
+			case "bytes_written":
+				attributes.Insert(metadata.A.Direction, pdata.NewAttributeValueString("sent"))
+				addToIntMetric(network, attributes, parseInt(v), now)
 			case "get_hits":
-				addIntSum(ilm.Metrics(), metadata.M.MemcachedGetHits.Init, now, parseInt(v))
+				attributes.Insert(metadata.A.Operation, pdata.NewAttributeValueString("get"))
+				statSlice := stats.Stats
+				hits := parseFloat(statSlice["get_hits"])
+				misses := parseFloat(statSlice["get_misses"])
+				if hits+misses > 0 {
+					addToDoubleMetric(hitRatio, attributes, (hits / (hits + misses) * 100), now)
+				} else {
+					addToDoubleMetric(hitRatio, attributes, 0, now)
+				}
+				attributes.Insert(metadata.A.Type, pdata.NewAttributeValueString("hit"))
+				addToIntMetric(operationCount, attributes, parseInt(v), now)
```

Comment:
What is going on here?

In some cases, we are calculating a hit ratio, but in others we are just using a count? These are two entirely different things and cannot be the same metric.